File: app/src/main/java/ru/geekbrains/notes/note/DateCreateSorterComparator.java
package ru.geekbrains.notes.note;

import java.util.Comparator;

public class DateCreateSorterComparator implements Comparator<Note> {
    @Override
    public int compare(Note note1, Note note2) {
        return Long.compare(note1.getDateCreate(), note2.getDateCreate());
    }
}

File: app/src/main/java/ru/geekbrains/notes/note/comparator/DateCreateSorterComparator.kt
package ru.geekbrains.notes.note.comparator

import ru.geekbrains.notes.note.Note
import java.util.*

class DateCreateSorterComparator : Comparator<Note> {
    override fun compare(note1: Note, note2: Note): Int {
        return java.lang.Long.compare(note1.dateCreate, note2.dateCreate)
    }
}